### PR TITLE
Run tests inside librecores CI docker image

### PR DESCRIPTION
**Pull request by Nancy-Chauhan**
Solves issue #81

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
@Nancy-Chauhan please use mentions to request reviews

**Comment by oleg-nenashev** [comment]
@stffrdhrn Hi. Did you have a chance to configure CI for your fork?

**Comment by stffrdhrn** [comment]
@oleg-nenashev
Do you mean with Travis? Yes, the link is here, I am not sure if it's public 
https://travis-ci.org/stffrdhrn/mor1kx/branches

I haven't tried anything on librecores-ci yet.

File: .travis.yml [hunk 1]
```diff
@@ -16,12 +19,11 @@ addons:
     - python3-pip
 
 before_install:
-  - sudo pip3 install --upgrade pip setuptools
-  - sudo pip3 install --ignore-installed six
-  - sudo pip3 install fusesoc
+  -  docker pull librecores/librecores-ci-dev
```
Comment (oleg-nenashev) [inline .travis.yml]: You are using wrong image. https://hub.docker.com/r/librecores/librecores-ci/ should be used

File: .travis.yml [hunk 2]
```diff
@@ -3,6 +3,9 @@ cache: ccache
 sudo: true
 dist: trusty
 
+services:
+  - docker 
+
 addons:
```
Comment (oleg-nenashev) [inline .travis.yml]: I doubt it is needed after switching to the Docker image

**Comment by Nancy-Chauhan** [comment]
CC @oleg-nenashev @stffrdhrn Can you please have a look. This seems to work with librecores-ci-image , but fails for verliator job and ESPRESSO pipeline.

**Comment by stffrdhrn** [comment]
Let's keep it in parallel. Instead of having the test.sh script that runs everything. Just use stages and run a different docker cmd and have different environment variables per run.

**Comment by stffrdhrn** [comment]
@oleg-nenashev @Nancy-Chauhan 
In addition to the last comment.  I've reason we use stages it to have a fall fast pipeline.  If verilator checks fail we don't want to go onto the regression testing.  Just fail fast.

Also, note we have marked espresso as known to fail.

With the test.sh we can't do either of those.

**Comment by Nancy-Chauhan** [comment]
@stffrdhrn @oleg-nenashev  Can you please check the recent changes
Travis log : https://travis-ci.com/Nancy-Chauhan/mor1kx/builds/114753627

**Comment by stffrdhrn** [comment]
> @stffrdhrn @oleg-nenashev Can you please check the recent changes
> Travis log : https://travis-ci.com/Nancy-Chauhan/mor1kx/builds/114753627

This looks great! If you are done I can merge this.

File: .travis/Dockerfile [hunk 3]
```diff
@@ -12,7 +12,7 @@
 #  'RUN .travis/install-verilator.sh' to this dockerfile to have an image
 #  with the full environment already installed.
 #
-FROM ubuntu:trusty
+FROM librecores/librecores-ci:latest
```
Comment (stffrdhrn) [inline .travis/Dockerfile]: It we run librecores/librecores-ci image in our tests do we need this `.travis/Dockerfile` anymore?  Do we use this file?

**Comment by Nancy-Chauhan** [comment]
@oleg-nenashev  @stffrdhrn  I think it is done . But Stafford is the verilator log fine to you  ? Because librecores-ci uses the recent version

File: .travis/Dockerfile (hunk 3 above)
Comment (Nancy-Chauhan) [reply .travis/Dockerfile]: @stffrdhrn This dockerfile does the same work what it used to do before i.e to simulate the travis environment only for testing purposes,this is not actually used in travis . And it works fine Librecores-ci base image

Comment (stffrdhrn) [reply .travis/Dockerfile]: Yes, but we needed to simulate the Travis environment to have something consistent for running tests. Now the tests run in docker so it's consistent.  This is not needed. Anyway I can remove.

**Comment by oleg-nenashev** [comment]
I was a bit late to the party. Thanks @Nancy-Chauhan, it is great to see CI running in the repo again !

**Comment by stffrdhrn** [comment]
I just merged because it's working. We can always do more changes.  We probably also want this change on the or1k_marocchino core.

**Comment by Nancy-Chauhan** [comment]
Yes @stffrdhrn, Can you please give more reference on or1k_marocchino core.

**Comment by stffrdhrn** [comment]
https://github.com/openrisc/or1k_marocchino?files=1

If you see the Travis file you will notice it's very similar.

**Comment by Nancy-Chauhan** [comment]
@oleg-nenashev @stffrdhrn Yes I see it is very similar. Will open PR soon.

**Pull request merged**